### ISSUE TITLE
Installation problem in Ubuntu 12.04

### ISSUE THREAD

**Issue by galfaroi** [lfex/lsci#39]
This happens at try to $make
I have the rebar function there any clues?

$ make
Getting dependencies ...
/bin/sh: 1: rebar: not found
make: *** [get-deps] Error 127

Thanks

**Comment by oubiwann** [comment]
Hrm ... you need the rebar executable file. Do you have rebar in your path?

What is the output of "which rebar" -- could you paste that?

I'm not sure if Ubuntu 12.04 had rebar in its apt archives, but if it did
and you installed it that way, you probably want to uninstall it and use
this instead:

$ wget https://github.com/rebar/rebar/wiki/rebar
$ chmod 755 rebar
$ sudo mv rebar /usr/local/bin

Let me know how that goes ...

**Comment by galfaroi** [comment]
Hi Duncan,

Thank you for reply. Now I have the following issue.

...
See git-pull(1) for details.

    Updating ./deps/py ...

Already up-to-date.
Cleaning ebin dir ...
Compiling project code and dependencies ...
cp: cannot create regular file `./ebin/lsci.app': No such file or directory
make: *** [proj-compile] Error 1


I realized I don't have installed Python 3, that could be the issue?

Thanks

**Comment by oubiwann** [comment]
You will defnitely need Python 3 installed, but that's not this particular
issue.

The update shouldn't be happening -- I'll have edit the Makefile to take
that out. But that's an operation that shouldn't err out the make
execution. Actually, the same was true for the first error you reported:
the make execution should have continued and ignored that error.

Hrm, do you have an environment variable set  for make? What is the output
of "echo $MAKEFLAGS"? It sounds like you or something you have installed is
giving make the "--stop" parameter. (Many of the make targets for LFE
projects use the "-" before commands to ignore errors and continue to the
best of their ability, usually when the error to be thrown is no bid deal).

Let me know, and I'll also look at things at this end ...

**Comment by oubiwann** [comment]
Hrm, it might be an alias set up, too. Perhaps something created an alias
for make that is "make --stop".

To find out, you can do this in bash:

  type make

And see what it returns.

**Comment by galfaroi** [comment]
The same as before 

this is what I got when :
$ type make
make is hashed (/usr/bin/make)
I don't see your update here in the repository do you forgot to push? I already installed Python 3 (but I had numpy in 2.7)

Already up-to-date.

	Updating ./deps/goldrush ...

You are not currently on a branch, so I cannot use any
'branch.<branchname>.merge' in your configuration file.
Please specify which branch you want to merge with on the command
line and try again (e.g. 'git pull <repository> <refspec>').
See git-pull(1) for details.

	Updating ./deps/kla ...

You are not currently on a branch, so I cannot use any
'branch.<branchname>.merge' in your configuration file.
Please specify which branch you want to merge with on the command
line and try again (e.g. 'git pull <repository> <refspec>').
See git-pull(1) for details.

	Updating ./deps/lager ...

You are not currently on a branch, so I cannot use any
'branch.<branchname>.merge' in your configuration file.
Please specify which branch you want to merge with on the command
line and try again (e.g. 'git pull <repository> <refspec>').
See git-pull(1) for details.

	Updating ./deps/lfe ...

You are not currently on a branch, so I cannot use any
'branch.<branchname>.merge' in your configuration file.
Please specify which branch you want to merge with on the command
line and try again (e.g. 'git pull <repository> <refspec>').
See git-pull(1) for details.

	Updating ./deps/ltest ...

You are not currently on a branch, so I cannot use any
'branch.<branchname>.merge' in your configuration file.
Please specify which branch you want to merge with on the command
line and try again (e.g. 'git pull <repository> <refspec>').
See git-pull(1) for details.

	Updating ./deps/lutil ...

You are not currently on a branch, so I cannot use any
'branch.<branchname>.merge' in your configuration file.
Please specify which branch you want to merge with on the command
line and try again (e.g. 'git pull <repository> <refspec>').
See git-pull(1) for details.

	Updating ./deps/py ...

Already up-to-date.
Cleaning ebin dir ...
Compiling project code and dependencies ...
cp: cannot create regular file `./ebin/lsci.app': No such file or directory
make: *** [proj-compile] Error 1

**Comment by oubiwann** [comment]
On Mon, Jan 12, 2015 at 8:59 PM, German Alfaro <notifications@github.com>
wrote:

> The same as before
>
> this is what I got when :
> $ type make
> make is hashed (/usr/bin/make)
> I don't see your update here in the repository do you forgot to push?
>

Nope. Hadn't gotten to it yet. Just did.


> I already installed Python 3 (but I had numpy in 2.7)
>

To ensure maximum ease of support, lsci expects that you will let it
install a local Python virtual env, per the instructions on the README:

$ make
$ . ./python/.venv/bin/activate

As such, you won't need to use (or worry about affecting) your system
install or already having NumPy, SciPy, etc., set up. (I'm guessing that
was clear, and you were just giving me extra info above... but better to be
safe with more info than sorry with less :-)).

Already up-to-date.
>
> [snip]

> Cleaning ebin dir ...
> Compiling project code and dependencies ...
> cp: cannot create regular file `./ebin/lsci.app': No such file or directory
> make: *** [proj-compile] Error  1
>
Okay, I did a fresh install myself, and ran into this same issue. I think I
know what it is ... I'll poke around, and send a long an update.

**Comment by oubiwann** [comment]
Okay, that did it (missing line in one of the make includes; didn't notice
sine the ./ebin directory had been created earlier on during project
development).

Thanks for submitting the bug!

Try it again after a git pull --all ...

**Comment by oubiwann** [comment]
Hrm, hang on a sec -- don't update yet.

Looks like there's been a ton of work done on ErlPort in the last week or
so (hurray!), but this introduces some breaking changes.

I will update the dependencies so that it pulls from a known good
configuration. You'll need to rm -rf ./deps/erlport and then re-run make.
I'll let you know when.

**Comment by oubiwann** [comment]
Okay!

That worked. I needed to pin ErlPort in LFE py (not lsci), so you'll need
to do the following in your lsci cloned dir:

$ rm -rf deps/py deps/erlport
$ make
$ . python/.venv/bin/activate
$ make repl-no-deps

Then, if you don't see errors after starting up, you should be good to go.
Once at the LFE REPL prompt, make sure you can call into Python, e.g.:

> (lsci-math:pi)
3.141592653589793

**Comment by galfaroi** [comment]
made some progress after struggling with a bug related to using venv in
python3 in Ubuntu uff (2.7 is very straightforward)
now I have this error:

Compiled src/lsci-util.lfe
Compiled src/lsci-py.lfe
Compiled src/lsci-asciiplot.lfe
Compiled src/lsci-stats.lfe
Compiled src/lsci-sp.lfe
Compiled src/lsci.lfe
Compiled src/lsci-frac.lfe
Compiled src/lsci-cmath.lfe
Compiled src/lsci-math.lfe
Compiled src/lsci-np.lfe
. python/.venv/bin/activate && \
    pip3 install -r python/requirements.txt
/bin/sh: 1: .: Can't open python/.venv/bin/activate
make: *** [get-py-deps] Error 2


On Mon, Jan 12, 2015 at 10:11 PM, Duncan McGreggor <notifications@github.com

**Comment by oubiwann** [comment]
Sad news, my friend: your distro may be too old for lsci :-(

As far as I'm aware, Ubuntu 12.04 has packages for Python 3.2. The
Python-native venv support didn't land until 3.3. If you can find more
recent Python packages for 12.04 (ideally, Python 3.4), you may be in luck.
Otherwise, you'll need to run with something more modern :-/

**Comment by galfaroi** [comment]
Yes, I manually update to Python3.3. pip3 and venv but still the same, I guess this would be the same until I upgrade. Do you think if only use erlport to load a Python function would be the same issue, I am new to Erlang, I would be awesome to have the same things at Python. Well to be honest Elixir is the thing to take to research this, I see it very haskelian, in a friendlier way..
